app/Helpers/database/achievement.php

```diff
@@ -165,10 +165,12 @@ function UploadNewAchievement(
     $consoleID = $gameData['ConsoleID'];
     $consoleName = $gameData['ConsoleName'];
     $isEventGame = $consoleName == 'Events';
-    $userPermissions = getUserPermissions($author);
+
+    $author = User::where('User', $authorUsername)->first();
+    $authorPermissions = (int) $author->getAttribute('Permissions');
```

Comment:
Recommend using `?->` or `??` in case `authorUsername` doesn't exist.

Reply:
In latest, adjusted to `?->` and changed the ORM query to use `firstWhere()`.

Reply:
Unrelated heads up: the reason that we have to access the permissions attribute like this instead of just `$author->Permissions` is because of the existing `->permissions()` relationship which causes a conflict. This will resolve itself as soon as the migration to roles is complete and this column removed.